hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java

```diff
@@ -465,11 +464,14 @@ public void assignedDistributionSet() {
 
         final long current = System.currentTimeMillis();
         final List<Target> targets = deploymentManagement.offlineAssignedDistributionSet(ds.getId(), controllerIds)
-                .getAssignedEntity();
+                .getAssignedEntity().stream().map(Action::getTarget).collect(Collectors.toList());;
         assertThat(actionRepository.count()).isEqualTo(20);
         assertThat(actionRepository.findByDistributionSetId(PAGE, ds.getId())).as("Offline actions are not active")
                 .allMatch(action -> !action.isActive());
-        assertThat(targetManagement.findByInstalledDistributionSet(PAGE, ds.getId()).getContent()).containsAll(targets)
+
+        assertThat(targetManagement.findByInstalledDistributionSet(PAGE, ds.getId()).getContent())
+                .usingElementComparator(controllerIdComparator())
```

Comment:
using `controllerIdComparator` because otherwise the `containsAll` fails due to object version mismatch